### DISCUSSION TITLE
Enhanced Mixer Board ideas

### DISCUSSION THREAD

**Discussion by SteveDaulton** [audacity/audacity#4224] [Ideas & Feature Requests]
### Your idea

Various features to enhance the Mixer Board.

![vertical-tracks](https://user-images.githubusercontent.com/11681097/186140676-d3374e34-100f-4847-a040-fdcae764bc10.png)

### Problem to be solved

Audacity's Mixer Board is very basic / primitive. At the very least it will need to support real-time effects.

**Comment by LWinterberg** [comment]
each item to be added to the meter should solve a problem: 

* An **effects stack** inside of the mixer will make it easier to apply and compare effect chains between tracks. 
* A **gain knob** would allow to non-destructively amplify a track pre-effects (with the volume slider applying post-effects)
* When multi-track-recording becomes a thing, 
    * an **Arm button** per track here lets you quickly arm and disarm many tracks
    * a button to **enable monitoring** (or software playthrough as it currently called) per track would be useful
    * an input source per track would be required

It's not yet entirely clear to me if "advanced" DAW things like Freezes, Sends, automation read/write modes, different output targets, arbitrary routing etc. are something Audacity really should have. 
It also isn't clear to me what benefit a vertical waveform would bring if a meter is in the mixer anyway.

**Comment by SteveDaulton** [comment]
> It also isn't clear to me what benefit a vertical waveform would bring if a meter is in the mixer anyway.

When a project has more than a few tracks, unless using a huge display it may not be possible to view the mixer and the normal track view at the same time.

It is common in other DAWs to display per-channel levels with vertical meters. Displaying the waveform is like a horizontal meter; less precise than a vertical meter, but enables you to look into the past and the future rather than just the instantaneous level.

How this could be beneficial is that when mixing a project, if you see there's big peaks visible ahead, you know not to mix too hot because it will clip when you reach the peaks.

Also note, it's just an idea, though one that I think could be useful as a non-default option. I'd have conventional, vertical peak meters, (or peak + RMS), as the default metering rather than waveforms.

**Comment by Paul-Licameli** [comment]
Solving the problem of the rendering of the stacked effects on a track during export, in 3.2, was the occasion for some righteous refactoring of internals that began to lay the foundations for audio graphs.  Sources and sinks of sample data and transformation stages in between.

An idea was also floated about insertion of meters for the metering of levels of the intermediate stages of transformation in effect stacks.

Is any of this adjacent to these ideas?

I could enjoy taking the refactoring further which would make implementation of such ideas easier.
